Source/WebCore/rendering/RenderBoxModelObject.cpp

```diff
@@ -322,37 +322,39 @@ DecodingMode RenderBoxModelObject::decodingModeForImageDraw(const Image& image,
     // A PaintBehavior may force synchronous decoding.
     if (paintInfo.paintBehavior.contains(PaintBehavior::Snapshotting))
         return DecodingMode::Synchronous;
-    if (paintInfo.paintBehavior.contains(PaintBehavior::ForceSynchronousImageDecode))
-        return DecodingMode::Synchronous;
 
-    // <img decoding="sync"> forces synchronous decoding.
+    auto isFlickeringPossible = [&]() -> bool {
+        // Not first paint, so we have to avoid flickering anyway.
+        if (!element() && element()->hasEverPaintedImages())
```

Comment:
The `!` seems wrong.

Reply:
Yes this should be 
```
   if (!(element() && element()->hasEverPaintedImages()))
            return false;
```
Since this means the element has never been drawn so it is safe to say async image decoding will not cause flickering.